### ISSUE TITLE
Fix some issues with l10n files

### ISSUE THREAD

**Issue by tukusejssirs** [pejulian/romcal#295]
@emagnier, you made some l10n issues when creating `v3`. Nothing serious, but still. Nevertheless, I’ll fix some other l10n issues too when creating the PR for this issue.

I’d like to fix them, but need you to confirm some stuff. Do tell me if you have some special requirements.

Anyway, no need to comment on of this, @emagnier, just go through it and add freely comment the suggested changes. If you agree with the proposed change, you can skip them; otherwise tell me what you think. Thanks! :pray:

Sorry for writing such a long issue description, but I wanted to be thorough. :wink:

---

1. I have never seen `º` used in English, therefore I sugest to use ordinary numbers. Check if the suggestion below is correct.

```js
// Original
octave: '{{count}}º Day within the Octave of the Nativity of the Lord',

// Suggested
octave: '$t(ordinals:{{count}}, capitalize) Day within the Octave of the Nativity of the Lord',
```

2. I think that all keys in `names` should be sorted in all locales.

3. I think we should remove `seasons` object from `en-gb.ts`. I know, I introduced this change, but now I think the changes in the `seasons` object does not matter. Even more: I think that the English Roman Missal (ed. typ. tertia) published in 2011 united the differences between the US and the UK terminology.

4. I’ll add `ordinals` to `sk.ts` and update the strings in `seasons` which use ordinal numbers.

5. I’ll proof `sk.ts`, as I have already found an issue (`Druhú nedeľa po narodení Pána` should be `Druhá nedeľa po narodení Pána`).

6. I’ll add all added keys to all locales.

7. Add the [l10n rules](https://github.com/romcal/romcal/issues/175#issuecomment-673048460) which are used to create the l10n keys to the docs.

8. Update the titles:
- `founder`/`foundress`;
- `king`/`queen`;
- `emperor`/`empress`;
- `pilgrim`.

9. We agreed to name the l10n keys according to [these rules](https://github.com/romcal/romcal/issues/175#issuecomment-673048460), therefore there are some issues with the following keys you added:

a. Missing canonisation level in the key value and title in the key name.

```js
adelaide_of_burgundy: 'Adelaide of Burgundy, Empress',
```

b. Missing surname; he was `Adelphus`, not `Adolph`.

```js
adelphus_of_metz_bishop: 'Saint Adolph, Bishop',
```

c. Remove `the` before `Saints`. In all `All Saints` (even when we mean just all saint of a particular region), we (usually) mean all saints, even unknown, therefore from grammatical point of view, we should not use the definite article (`the`).

As for the dioceses, the `Diocese of Paris` and the `Diocese of Saint-Denis` are considered as proper names, therefore we should use capital `D` in `Diocese`.

Actually, Paris is an archdiocese, isn’t it? Should it be renamed to `Archdiocese`?

```js
all_the_saints_of_the_diocese_of_paris: 'All the Saints of the diocese of Paris',
all_the_saints_of_the_diocese_of_saint_denis: 'All the Saints of the diocese of Saint-Denis',
```

d. In English, we usually don’t use `First Bishop of`, not even `Bishop of`, therefore I suggest to change it to `Saint Amandus of Strasbourg, Bishop` (key: `amandus_of_strasbourg_bishop`).

Also I think that the patronages of dioceses should always include the full name of the diocese (e.g. `Patron of the Diocese of Paris`). If you agree, I’ll update the [rules](https://github.com/romcal/romcal/issues/175#issuecomment-673048460) accordingly.

Thirdly, should we state two patronages in one key (`Patron of the city` and `the Diocese of Saint-Denis`)? I have a similar patronage in one of the Slovak dioceses, but I was thinking rather to create two separate keys, one used in the diocesan calendar, the other in the calendar of the city/town. Tell me what do you think.

```js
amandus_first_bishop_of_strasbourg: 'Saint Amand, First Bishop of Strasbourg',
denis_of_paris_bishop_patron_of_the_diocese_of_paris: 'Saint Denis, Martyr, First bishop, Patron of the diocese',
denis_of_paris_bishop_patron_of_the_diocese_of_saint_denis: 'Saint Denis, Bishop and Martyr,Patron of the city and the diocese of Saint-Denis',
```

e. Missing surname; I have found that he was an abbot in a monastery around Clermont, maybe in the Vosges. Should we call him St Amarin of Clermont or rather St Amarin of Vosges?

```js
amarin_abbot: 'Saint Amarin, Abbot and Martyr',
```

f. Do you think that German `Strassburg` should be used instead of `Starsbourg`? I think we might use `Strassburg` because that might be be the name of the city at the time, but I still suggest to change it to Strasbourg.

```js
arbogast_of_strassburg_bishop: 'Saint Arbogast, Bishop',
arbogast_of_strassburg_bishop_patron_of_the_diocese_of_strasbourg: 'Saint Arbogast, Bishop, Patron of the diocese of Strasbourg',
```

g. Which Aurelia is this? According to the Wiki, both are celebrated on 15 October. You don’t even state the date of death in `fr.ts`. I think is is Aurelia of Strasbourg, because it is celebrated in `france.strasbourg.ts`.
- Aurelia of Strasbourg († 4th c.) https://en.wikipedia.org/wiki/Aurelia_of_Strasbourg
- Aurelia of Regensburg († 1027) https://en.wikipedia.org/wiki/Aurelia_of_Regensburg

```js
aurelia_virgin: 'Saint Aurelia, Virgin',
```

h. In English, `Martyrs` should be written with capital initial letter (i.e. in title case)

```js
blessed_martyrs_of_paris: 'Blessed martyrs of Paris',
```

i. Is there any reason you decided to use `Religious` title in English? I could not find a reference with `Religious` in English, only with `Virgin`. Even in Latin (MR2004), she is called a virgin.

```js
catherine_laboure_virgin: 'Saint Catherine Labouré, Religious',
```

j. I suggest to change the name to `Ceraunus` (https://en.wikipedia.org/wiki/Ceraunus).

```js
ceran_of_paris_bishop: 'Saint Céran of Paris, Bishop',
```

k. I suggest we should agree upon the naming of the celebrations of dedications of churches (incl cathedrals). I think we should specify the official church names (localised if necessary) with the place (city/town) where they are located, e.g. `Notre-Dame Cathedral in Paris` or `Cathedral of Saint Denis in Paris`. If you agree, I’ll change the following key values as well as update the [rules](https://github.com/romcal/romcal/issues/175#issuecomment-673048460).

Actually, should `Cathedral of Saint Denis in Paris` be rather called a basilica ([src](https://en.wikipedia.org/wiki/Basilica_of_Saint-Denis))?

```js
dedication_of_the_cathedral_of_paris: 'Dedication of the Cathedral of Paris',
dedication_of_the_cathedral_of_saint_denis: 'Dedication of the Cathedral of Saint-Denis',
dedication_of_the_cathedral_of_strasbourg: 'Dedication of the Cathedral of Strasbourg',
```

l. In the following keys, I’d rather use `Sunday of Divine Mercy` and `Sunday of the Word of God` as subtitles (not a feature yet) to `Second Sunday of Easter`/`Third Sunday in Ordinary Time` celebration names, where a subtitle is additional property in the `LiturgicalDay`. However, if you really want (at least in the interim) we might keep the subtitle in the celebration name, but I suggest to change `or` to `or Sunday`.

```js
divine_mercy_sunday: 'Second Sunday of Easter, or of Divine Mercy',
sunday_of_the_word_of_god: 'Third Sunday in Ordinary Time, or of the Word of God',
```

m. Change the name to `Eligius`.

```js
eloi_of_noyon_bishop: 'Saint Eloi, Bishop',
```

n. `Trèves` in French is `Trier` in English

```js
eucharius_of_treves_bishop: 'Saint Eucharius, Bishop',
```

o. Why did you use `Saint` twice (even in French)? In romcal `v1`, this usually meant that the two persons are celebrated separately (as two optional memorials). If it is not the case here, I suggest to change the key value to `Saints Eugenie and Attala, Virgins` (even in French).

Change `Attale` to `Attala` ([src](https://catholicsaints.info/saint-attala-of-strasbourg/)).

Also the key should be changed to `eugenie_of_alsace_and_attala_of_alsace` as per the [rules](https://github.com/romcal/romcal/issues/175#issuecomment-673048460).

```js
eugenie_and_attale_of_alsace: 'Saint Eugenie et Saint Attale, Virgin',
```

p. I think we should add `founder`/`foundress` to the list of titles (see point 9 below), therefore (as per the rules), we should append `_founder` to the key name.

```js
frederic_ozanam: 'Blessed Frederic Ozanam, Founder',
```

q. Add `_of_autun`.

```js
leodegar_bishop_and_martyr: 'Saint Leodegar, Bishop and Martyr',
```

r. Change `of` to `de` (`Saint Louise de Marillac, Religious`).

```js
louise_of_marillac: 'Saint Louise of Marillac, Religious',
```

s. I suggest to change the key name to `ludan_of_scotland_pilgrim`

```js
ludan_pilgrim: 'Saint Ludan, Pilgrim',
```

t. Change the name to `Marcellus`

```js
marcel_of_paris_bishop: 'Saint Marcel of Paris, Bishop',
```

u. According to the rules, key name structure of religious people should include besides their religious name also their surname, therefore we should add `Guyart`.

According to Wiki, the English name of this blessed is `Marie of the Incarnation`.

You used a opening quote (`‘`) instead of an apostrophe (`’`, which is the same character as the closing quote), so this needs to be fixed in `fr.ts` and `en.ts`.

```js
marie_de_incarnation_religious: 'Blessed Marie de l‘Incarnation, Religious',
```

v. According to the rules, key name structure of religious people should include besides their religious name also their surname, therefore we should add `Smet`.

Although I could not find a reference, we might want to change her English name to `Marie of the Providence` (as it is with `Marie of the Incarnation Guyart`).

```js
marie_de_la_providence_virgin: 'Blessed Marie de la Providence, Virgin',
```

w. Add `de_jesus` to the key name.

Change `virgin` to `religious` in the key name.

Add surname `Milleret` to the English key value.

Use `Foundress` title instead of `Founder` in the key value.

```js
marie_eugenie_milleret_virgin: 'Saint Marie-Eugénie de Jésus, Religious and Founder',
```

x. I presume that this person is same as [this](https://en.wikipedia.org/wiki/Sophie-Th%C3%A9r%C3%A8se_de_Soubiran_La_Louvi%C3%A8re), but there are two differences:
- `Marie-Thérèse` is changed to `Sophie-Thérèse` (first name); her religious name is `Marie of the Sacred Heart`;
- she died on 7 June 1893.

If my assumption is correct, consider using the name `Mary of the Sacred Heart de Soubiran`.

```js
marie_therese_de_soubiran_virgin: 'Blessed Marie-Thérèse de Soubiran, Virgin',
```

y. It should be `Our Lady, Refuge of Sinners`.

```js
mary_refuge_of_sinner: 'Holy Mary, Refuge of Sinner',  
```

z. In English, there is no space before semicolon.

Remove the first `_and` from the key name.

Replace `treves` with `trier` in the key name.

Replace `Valere` with `Valerius` in the key name.

Replace `Euchaire` with `Eucharius` in the key name.

Add `Bishops` title.

IMO we should use either `maternus_of_cologne_valerius_of_treves_and_eucharius_bishops` or `first_apostles_of_alsace` as the key name.

Change the key value to either `First Apostles of the Church of Alsace: Saints Maternus, Valerius and Eucharius, Bishops` or `Saints Maternus, Valerius and Eucharius, Bishops, the First Apostles of the Church of Alsace` or simply `Saints Maternus, Valerius and Eucharius, Bishops`.

```js
maternus_of_cologne_and_valerius_of_treves_and_eucharius_first_apostles_of_alsace: 'The first Apostles of the Church of Alsace : Saints Maternus, Valere and Euchaire',
```

aa. Change `Merry` to `Mederic` and `Droctovée` to `Droctoveus` respectively.

Remove redundant `Saint` (and change the first one to `Saints`).

Change the key name to `mederic_of_autun_and_droctoveus_of_autun_abbots`.

```js
merry_and_saint_droctovee_abbots: 'Saint Merry and Saint Droctovée, Abbots',
```

ab. Change `Modeste` to `Modest` and `André` to `Andrew`.

```js
modeste_andlauer_and_andre_bauer_martyrs: 'Saints Modeste Andlauer and André Bauer, Martyrs',
```

ac. Change `Modeste` to `Modest`.

```js
modeste_andlauer_martyr: 'Saint Modeste Andlauer, Martyr',
```

ad. Change `Morandus` to `Morand`.

```js
morandus_of_cluny_monk: 'Saint Morandus, Monk',
```

ae. Add `priest` to the key name.

Actually, in `france.paris.ts`, do they celebrated him only with `priest` title? He was a religious, a priest and a founder.

```js
nicholas_barre: 'Blessed Nicholas Barré, Priest',
```

af. Add `and ` before `Patroness`.

```js
odile_of_alsace_abbess_patroness_of_alsace: 'Saint Odile, Abbess, Patroness of Alsace',
```

ag. Change `Ouen` to `Audoin`. Add `_of_rouen` to the key name.

```js
ouen_bishop: 'Saint Ouen, Bishop',
```

ah. Append `_empress` to the key name

```js
richardis_of_swabi: 'Saint Richardis, Empress',
```

ai. Consider changing the key name to `thursday_of_the_lords_supper`.

Consider moving the key to the `seasons` object.

```js
thursday_of_the_lord_s_supper: 'Thursday of the Lord’s Supper',
```

aj. Change `pape` to `pope`.

Consider removing `and Martyr`, as he was not a martyr (even MR2004 states that he _faithfully guided the Roman Church for eight years_ after the martyrdom of St Callistus).

```js
urban_i_pape: 'Saint Urban I, Pope and Martyr',
```

ak. Replace `treves` with `trier` in the key name.

```js
valerius_of_treves_bishop: 'Saint Valerius, Bishop',
```

al. @emagnier, you added `Hermit` title; it is celebrated so in `france.strasbourg.ts`? It is also celebrated in [Germany](https://www.eucharistiefeier.de/lk/api/lko0000.html) as `Wendelin, Einsiedler im Saarland` (`Wendelin, Hermit in Saarland`) and [Wiki](https://en.wikipedia.org/wiki/Wendelin_of_Trier) states that he was a hermit, and only some legends say he was later consecrated abbot.

```js
wendelin_of_trier_abbot: 'Saint Wendelin, Hermit and Abbot',
```

**Comment by emagnier** [comment]
**7.** `Add the l10n rules which are used to create the l10n keys to the docs.`

Could it be a good idea to gather these rules in a wiki or a .md file? 
For simplicity, we could start in a wiki, and then move this documentation in md files when we are ready.

**9.** Yes, adding these new calendars took me a lot of time, and I sometimes worked late on it. Thank you for taking the time of a deep review.

**c.** `Actually, Paris is an archdiocese, isn’t it? Should it be renamed to Archdiocese?`

Yes it is, technically. But in practice, we only say/write "Diocese of Paris".

**e.** It's also Amarinus in Latin. I suggest amarin_of_alsace_abbot, or amarinus_of_alsace_abbot

**f.** I prefer `Strasbourg` which is the right spelling in french.

**g.** Yes the first one makes sense. I didn't had too much information on this saints.

**k.**

In the diocese of Paris, even in English the name of the Cathedral is "Notre-Dame de Paris".
There is no mentions of Saint Denis in its name.

In the diocese of Saint Denis, indeed it's also called as a basilica, but in the context of this liturgical day, we are especially celebrating the feast / solemnity of the "main" church of the diocese, where the bishop's cathedra is placed. So here there are good reasons to keep "Cathedral", or possibly Basilic-Cathedral. 

**i.**

Yes, I like the idea of subtitles. Because we keep a standard main title (and key), while giving additional content (in a separated field) to this liturgical day.

**al.** "Hermit" is the title given in the proper missal of the diocese of Strasbourg.

---

Tell me know if you think I've missed something. Thanks!

**Comment by tukusejssirs** [comment]
> **7.** `Add the l10n rules which are used to create the l10n keys to the docs.`
>
> Could it be a good idea to gather these rules in a wiki or a .md file?
> For simplicity, we could start in a wiki, and then move this documentation in md files when we are ready.

Actually, I think that in the end we should create a function from the rules, once we have completed more or less the following points:
- create/use a localisation tool (#43);
- when a user adds a new celebration/person, a new key would be generated, but until someone aproves the addition, the key should be editable (i.e. for example when a person’s name changes, the key would be updated); once aproved, the key name should not change (a key name change is permittable when creating a new major version).

I think that the rules should be part of the dev docs (as opposed to the user docs). That said, it could be in the GitHub Wiki (at least for now), but the rules _might_ change in the future (not couting additions); then it makes sense to version them.

I’ll leave it up to you.

> **9.**
>
> **c.** `Actually, Paris is an archdiocese, isn’t it? Should it be renamed to Archdiocese?`
>
> Yes it is, technically. But in practice, we only say/write "Diocese of Paris".

Well, even in Slovakia we _informally_/_colloquially_ talk about archdioceses as dioceses, but not in the formal conversations and/or official books. As we strive to be accurate, I suggest to add this to the rules to always use the proper name of the diocese.

> **e.** It's also Amarinus in Latin. I suggest amarin_of_alsace_abbot, or amarinus_of_alsace_abbot

Okay. I think we agree upon `amarin_of_alsace_abbot: 'Saint Amarin, Abbot and Martyr'`.

> **f.** I prefer `Strasbourg` which is the right spelling in french.

Okay, `Strasbourg` it is. :wink:

> **g.** Yes the first one makes sense. I didn't had too much information on this saints.

I am not sure if I got you right. Do you merely _think_ it is Aurelia of Strasbourg (probably based on `of Strasbourg`) or are you sure? Your last sentence suggests you aren’t sure at all (just like me). :wink: Is there a way you could get some Ordo/Missal/anything that contains some info about her? Although it makes sense and even [Wiki](https://en.wikipedia.org/wiki/Saint_Aurelia%27s_Church,_Strasbourg#The_name) says that there was/is a cult of Aurelia of Strasbourg explicitly.

> **k.**
>
> In the diocese of Paris, even in English the name of the Cathedral is "Notre-Dame de Paris".

Well, I understand what you mean (even [Wiki](https://en.wikipedia.org/wiki/Notre-Dame_de_Paris) uses `Notre-Dame de Paris`), when search the web, `Notre-Dame Cathedral` can be found quite often on English websites. Also, the very same Wiki article states that the official name is `Cathédrale Notre-Dame de Paris` (litteraly in English `Our Lady of Paris Cathedral`).

Also note that in English, the patronage of church (`Notre-Dame [de Paris]`) _usually_ precedes the type of church (`Cathedral`).

I don’t say that in English, the French name is not used, but I think that the names of churches are _usually_ translated. I understand that Notre-Dame Cathedral is one of the most famous churches in the world, therefore I am _not_ against using `Notre-Dame` instead of `Our Lady`, but I still suggest to use `Notre-Dame [of Paris] Cathedral` and `Cathédrale Notre-Dame [de Paris]` instead of `Notre-Dame de Paris`.

What do you think?

> There is no mentions of Saint Denis in its name.

That was about `dedication_of_the_cathedral_of_saint_denis`, not about `dedication_of_the_cathedral_of_paris`.

> In the diocese of Saint Denis, indeed it's also called as a basilica, but in the context of this liturgical day, we are especially celebrating the feast / solemnity of the "main" church of the diocese, where the bishop's cathedra is placed. So here there are good reasons to keep "Cathedral", or possibly Basilic-Cathedral.

I see. Its [website](http://www.saint-denis-basilique.fr/en/#) uses `Basilique-cathédrale de Saint-Denis`, therfore I suggest to use it in French. Likewise, in English I suggest to use `Basilica-Cathedral of Saint-Denis` (if talking about a city/place; or `… of Saint Denis` if talking about a saint).

I didn’t emphesise it and you didn’t comment on `dedication_of_the_cathedral_of_strasbourg`. According to [Wiki](https://en.wikipedia.org/wiki/Strasbourg_Cathedral), its official name is `Cathedral of Our Lady of Strasbourg` (or `Our Lady of Strasbourg Cathedral`; French name is `Cathédrale Notre-Dame de Strasbourg`)

> **al.** "Hermit" is the title given in the proper missal of the diocese of Strasbourg.

Okay. Even MR2004 titles him only a hermit, therefore I suggest to to use only the `Hermit` title and change the key name: `wendelin_of_trier_hermit: 'Saint Wendelin, Hermit'`.

---

I’ll try to start working on this later this week (probably this weekend).

**Comment by tukusejssirs** [comment]
> **7.** `Add the l10n rules which are used to create the l10n keys to the docs.`
>
> Could it be a good idea to gather these rules in a wiki or a .md file?
> For simplicity, we could start in a wiki, and then move this documentation in md files when we are ready.

Actually, I think that in the end we should create a function from the rules, once we have completed more or less the following points:
- create/use a localisation tool (#43);
- when a user adds a new celebration/person, a new key would be generated, but until someone aproves the addition, the key should be editable (i.e. for example when a person’s name changes, the key would be updated); once aproved, the key name should not change (a key name change is permittable when creating a new major version).

I think that the rules should be part of the dev docs (as opposed to the user docs). That said, it could be in the GitHub Wiki (at least for now), but the rules _might_ change in the future (not couting additions); then it makes sense to version them.

I’ll leave it up to you.

> **9.**
>
> **c.** `Actually, Paris is an archdiocese, isn’t it? Should it be renamed to Archdiocese?`
>
> Yes it is, technically. But in practice, we only say/write "Diocese of Paris".

Well, even in Slovakia we _informally_/_colloquially_ talk about archdioceses as dioceses, but not in the formal conversations and/or official books. As we strive to be accurate, I suggest to add this to the rules to always use the proper name of the diocese.

> **e.** It's also Amarinus in Latin. I suggest amarin_of_alsace_abbot, or amarinus_of_alsace_abbot

Okay. I think we agree upon `amarin_of_alsace_abbot: 'Saint Amarin, Abbot and Martyr'`.

> **f.** I prefer `Strasbourg` which is the right spelling in french.

Okay, `Strasbourg` it is. :wink:

> **g.** Yes the first one makes sense. I didn't had too much information on this saints.

I am not sure if I got you right. Do you merely _think_ it is Aurelia of Strasbourg (probably based on `of Strasbourg`) or are you sure? Your last sentence suggests you aren’t sure at all (just like me). :wink: Is there a way you could get some Ordo/Missal/anything that contains some info about her? Although it makes sense and even [Wiki](https://en.wikipedia.org/wiki/Saint_Aurelia%27s_Church,_Strasbourg#The_name) says that there was/is a cult of Aurelia of Strasbourg explicitly.

> **k.**
>
> In the diocese of Paris, even in English the name of the Cathedral is "Notre-Dame de Paris".

Well, I understand what you mean (even [Wiki](https://en.wikipedia.org/wiki/Notre-Dame_de_Paris) uses `Notre-Dame de Paris`), when search the web, `Notre-Dame Cathedral` can be found quite often on English websites. Also, the very same Wiki article states that the official name is `Cathédrale Notre-Dame de Paris` (litteraly in English `Our Lady of Paris Cathedral`).

Also note that in English, the patronage of church (`Notre-Dame [de Paris]`) _usually_ precedes the type of church (`Cathedral`).

I don’t say that in English, the French name is not used, but I think that the names of churches are _usually_ translated. I understand that Notre-Dame Cathedral is one of the most famous churches in the world, therefore I am _not_ against using `Notre-Dame` instead of `Our Lady`, but I still suggest to use `Notre-Dame [of Paris] Cathedral` and `Cathédrale Notre-Dame [de Paris]` instead of `Notre-Dame de Paris`.

What do you think?

> There is no mentions of Saint Denis in its name.

That was about `dedication_of_the_cathedral_of_saint_denis`, not about `dedication_of_the_cathedral_of_paris`.

> In the diocese of Saint Denis, indeed it's also called as a basilica, but in the context of this liturgical day, we are especially celebrating the feast / solemnity of the "main" church of the diocese, where the bishop's cathedra is placed. So here there are good reasons to keep "Cathedral", or possibly Basilic-Cathedral.

I see. Its [website](http://www.saint-denis-basilique.fr/en/#) uses `Basilique-cathédrale de Saint-Denis`, therfore I suggest to use it in French. Likewise, in English I suggest to use `Basilica-Cathedral of Saint-Denis` (if talking about a city/place; or `… of Saint Denis` if talking about a saint).

I didn’t emphesise it and you didn’t comment on `dedication_of_the_cathedral_of_strasbourg`. According to [Wiki](https://en.wikipedia.org/wiki/Strasbourg_Cathedral), its official name is `Cathedral of Our Lady of Strasbourg` (or `Our Lady of Strasbourg Cathedral`; French name is `Cathédrale Notre-Dame de Strasbourg`)

> **al.** "Hermit" is the title given in the proper missal of the diocese of Strasbourg.

Okay. Even MR2004 titles him only a hermit, therefore I suggest to to use only the `Hermit` title and change the key name: `wendelin_of_trier_hermit: 'Saint Wendelin, Hermit'`.

---

I’ll try to start working on this later this week (probably this weekend).

**Comment by emagnier** [comment]
Go as you think it's best. I don't have a lot of time to dive into all of this.

Just a note about the "Diocese of Paris", it's written like this in liturgical books (proper missal & dictionary, even ordos). I didn't see any mention of "Archdiocese" even in official information from this diocese. This is exactly the same for Strasbourg, where there is also an Archbishop, but we only use the name "Diocese of Strasbourg".

And FYI, for the Diocese of Paris, Saint-Denis and Strasbourg, calendar / saints information are directly coming from official proper missal of these dioceses.

**Comment by tukusejssirs** [comment]
> Just a note about the "Diocese of Paris", it's written like this in liturgical books (proper missal & dictionary, even ordos). I didn't see any mention of "Archdiocese" even in official information from this diocese. This is exactly the same for Strasbourg, where there is also an Archbishop, but we only use the name "Diocese of Strasbourg".

I see. It is quite strange practice from my point of view. We in Slovakia use the (IMHO) proper naming even in the official books (Missal, lectionaries, Ordos, …).

I agree we should preferably use the naming used in the official books (and so we’ll do), although I don’t uderstand why it is so. I know that each archdiocese is also a diocese, but I’ve never thought that `diocese` is used as the main name for an archdiocese anywhere. :wink:

**Comment by tukusejssirs** [comment]
@emagnier, I noticed that in some languages (at least in `la.ts` and `es.ts`) for `ordinals` and/or `months` created  some additional keys, like for ordinal `1` in `la.ts`:

```js
'1': 'primum',
'1_masculine': 'primus',
'1_feminine': 'prima',
```

I think using [`i18next-fluent`](https://github.com/i18next/i18next-fluent) it would be much better, because the grammar of some languages is quite simple for our use case (e.g. English), others are more complicated (e.g. Latin, Slovak, Czech, …). I know you tried to minimise the romcal dependencies, but `i18next-fluent` might ease some l10n stuff.

For example, in Slovak for all ordinals (and possibly for other adjectives in the future), we need to define a string for each gender (masculine, feminine, neuter), each for both singular and plural (i.e. `3 × 2 = 6` strings per two strings in English [singular and plural]). For months, ranks and titles (i.e. nouns and pronouns), we need (at least) Nominative and Genitive for both singular and plural; note that usually we have two plural forms (i.e. for `1`, `2-4` and `0, 5+`), but other languages have different grammer rules.

Anyway, I see an issue in the Latin example above: why did you use `1` instead of `1_neuter`?

**Comment by tukusejssirs** [comment]
> 4. I’ll add `ordinals` to `sk.ts` and update the strings in `seasons` which use ordinal numbers.
>
> (Also related to [this comment of mine](https://github.com/romcal/romcal/issues/295#issuecomment-932912908).)

@emagnier, I need an advice.

In Slovak, every noun and pronoun is either masculine, feminine or neuter and based on this gender that word is inflected. I want to localise `ordinals`, which is not a problem, but the problem is that not every weekday name in Slovak has one gender, e.g. `nedeľa` (Sunday) is feminine, but `pondelok` (Monday) is masculine.

How could I make this work, e.g., in `seasons.advent.weekday`?

---

**Update:** I thinks we should create a separate issue for making it possible to define genders of `weekday` names (possibly as an `i18next` context) in order to use it choosing the right gender of `ordinals`. Then it’ll be quite easy to fix the point 4.

**Update 2:** As far as I understand the `i18next` docs correctly, there can be a single context for each l10n key, therefore we would need to combine all needed contexts (like `{gender} + {grammar_case} + {number}`, e.g. `masculine + nominative + singular`) whenever we need more than context, which will become overly complicated and less readable as we add support for other languages, not to mention that there would be a chance that it could be broken for other languages that have a particular context not defined. Therefore I strongly suggest to use [`i18next-fluent`](https://github.com/i18next/i18next-fluent) and thus [Fluent](https://projectfluent.org/).

**Comment by tukusejssirs** [comment]
> [@tukusejssirs]
>
> I agree we should preferably use the naming used in the official books (and so we’ll do), although I don’t uderstand why it is so. I know that each archdiocese is also a diocese, but I’ve never thought that `diocese` is used as the main name for an archdiocese anywhere. :wink:

@emagnier, just a quick question: what do think about using `diocèse de Paris` in French only and `Archdiocese of Paris` in all other locales, as well as in the key name itself?

**Comment by emagnier** [comment]
> just a quick question: what do think about using diocèse de Paris in French only and Archdiocese of Paris in all other locales, as well as in the key name itself?

Yes, I'm ok for that.

**Comment by tukusejssirs** [comment]
These subissues of #295 are moved to separate issues:

- points 4 and 11 are moved to #369;
- point 7 is moved to #370;
- point 8 is moved to #232;
- point 10 (d) is moved to #371;
- point 10 (i) is moved to #297;
- point 13 is moved to #136.

All other points will be fixed in #350.
